hack/update-openapigen.sh

```diff
@@ -22,8 +22,22 @@ if [ -z "${GOPATH:-}" ]; then
     export GOPATH=$(go env GOPATH)
 fi
 
+KNOWN_VIOLATION_EXCEPTIONS=hack/violation_exceptions.list
+CURRENT_VIOLATION_EXCEPTIONS=hack/current_violation_exceptions.list
+
 # Generating OpenAPI specification
-go run vendor/k8s.io/code-generator/cmd/openapi-gen/main.go --input-dirs github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha2,knative.dev/pkg/apis --output-package github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha2/ --go-header-file hack/boilerplate.go.txt
+go run vendor/k8s.io/kube-openapi/cmd/openapi-gen/openapi-gen.go \
```

Comment:
Is there a way to run this without using vendor considering we are going migrating to go mod and getting rid of vendor ?

Reply:
+1. Regardless of whether we get rid of vendor library or not, it would be good idea to start making the code logic independent of vendor dir.

Reply:
After migrate to go mod/kubebuilder2, the story would be big different. User need run ```go get XXX```  to get the package firstly, then configure the path of ```$(GOBIN)/openapi-gen``` and then can run above command(or we can check and then do it). 
I think the logic is part of kubebuilder2 scaffold for controller-gen and we just add all our requirements there(not only here but all places that require a dependency in code), so suggest we do that after migrate to kubebuilder2.